rust/index/src/hnsw_provider.rs

```diff
@@ -324,21 +382,21 @@ impl HnswIndexProvider {
             }
         };
 
-        match HnswIndex::load(index_storage_path_str, &index_config, *id) {
-            Ok(index) => {
-                let _guard = self.write_mutex.lock().await;
-                match self.get(id, cache_key).await {
-                    Some(index) => Ok(index.clone()),
-                    None => {
-                        let index = HnswIndexRef {
-                            inner: Arc::new(RwLock::new(index)),
-                        };
-                        self.cache.insert(*cache_key, index.clone()).await;
-                        Ok(index)
-                    }
+        let _guard = self.write_mutex.lock().await;
```

Comment:
nit: `self.write_mutex.lock().instrument(tracing::trace_span!(parent: Span::current(), "Mutex acquire for hnsw load")).await`